tests/conftest.py

```diff
@@ -62,7 +62,7 @@ def network():
     return _get_network()
 
 
-@pytest.fixture(scope='session')
+@pytest.yield_fixture(scope='session')
 def data_network():
```

Comment:
I think we can find a better name for this fixture as it is used now. Do you have any idea?

Reply:
Something to reflect its usage only in test_execution* files and its content ? What about `multi_node_execution_env` ?

Reply:
Sounds like a good idea. As it is the only execution environment, I will call it directly `execution_env` (or `global_execution_env`). `multi_node` could be confusing as it is also used in tests using a single node.

Reply:
`global_execution_env` it is!